src/lib/syncPagePath.ts

```diff
@@ -1,24 +1,36 @@
 import type { APIv2PageUploadCommands } from '../index.js';
 import type {
+  CustomPagesRequestRepresentation,
+  CustomPagesResponseRepresentation,
   GuidesRequestRepresentation,
   GuidesResponseRepresentation,
   ProjectRepresentation,
   ReferenceRequestRepresentation,
   ReferenceResponseRepresentation,
 } from './types/index.js';
 
+import path from 'node:path';
+
 import chalk from 'chalk';
 import ora from 'ora';
 import toposort from 'toposort';
 
 import { APIv2Error } from './apiError.js';
 import { oraOptions } from './logger.js';
-import { findPages, type PageMetadata } from './readPage.js';
+import { allowedMarkdownExtensions, findPages, type PageMetadata } from './readPage.js';
 import { categoryUriRegexPattern, parentUriRegexPattern } from './types/index.js';
 import { validateFrontmatter } from './validateFrontmatter.js';
 
-type PageRequestRepresentation = GuidesRequestRepresentation | ReferenceRequestRepresentation;
-type PageResponseRepresentation = GuidesResponseRepresentation['data'] | ReferenceResponseRepresentation['data'];
+type PageRequestRepresentation =
+  | CustomPagesRequestRepresentation
+  | GuidesRequestRepresentation
+  | ReferenceRequestRepresentation;
```

Comment:
nit: can you rearrange these types a bit so we can do something like this?

```suggestion
  | GuidesReferenceRequestRepresentation
```